src/AutoRest.CSharp/Mgmt/Output/MgmtObjectType.cs

```diff
@@ -108,6 +109,18 @@ protected virtual ObjectTypeProperty CreatePropertyType(ObjectTypeProperty objec
                     if (match != null)
                     {
                         propertyType = ReferenceTypePropertyChooser.GetObjectTypeProperty(objectTypeProperty, match);
+                        if (match.Implementation is SystemObjectType systemObjectType
+                            && systemObjectType.SystemType == typeof(ManagedServiceIdentity)
+                            && (typeToReplace.Properties.First(p => p.Declaration.Name == "Type")?.SchemaProperty?.Schema is SealedChoiceSchema sealedChoiceSchema
+                                && sealedChoiceSchema.Choices.Any(c => c.Value == ManagedServiceIdentityTypeV3Converter.systemAssignedUserAssignedV3Value)
+                                || typeToReplace.Properties.First(p => p.Declaration.Name == "Type")?.SchemaProperty?.Schema is ChoiceSchema choiceSchema
+                                && choiceSchema.Choices.Any(c => c.Value == ManagedServiceIdentityTypeV3Converter.systemAssignedUserAssignedV3Value)))
+                        {
+                            if (objectTypeProperty.SchemaProperty!.Schema.Extensions == null)
+                                objectTypeProperty.SchemaProperty!.Schema.Extensions = new DictionaryOfAny() { { "x-ms-mgmt-modelVersion", "v3" } };
+                            else
+                                objectTypeProperty.SchemaProperty!.Schema.Extensions?.TryAdd("x-ms-mgmt-modelVersion", "v3");
+                        }
```

Comment:
Could we add some comments to describe what this is doing?

Reply:
Added and refactored.